iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -589,13 +597,18 @@ public void processPushMessageReceiptEvent(PushMessageReceiptEvent event) throws
                 //Android and iOS have differently encoded payload formats. See the tests for examples.
                 if (event.getRequest().getRuntimeEnvironment() instanceof AndroidRuntimeEnvironment) {
                     Map<String, Object> iterableMap = mapper.readValue((String) payload.get("itbl"), Map.class);
-                    request.campaignId = Integer.parseInt(mapper.writeValueAsString(iterableMap.get("campaignId")));
-                    request.templateId = Integer.parseInt(mapper.writeValueAsString(iterableMap.get("templateId")));
+                    request.campaignId = convertItblPayloadFieldToInt(iterableMap.get("campaignId"));
+                    request.templateId = convertItblPayloadFieldToInt(iterableMap.get("templateId"));
                     request.messageId = (String) iterableMap.get("messageId");
                 } else {
-                    request.campaignId = Integer.parseInt(mapper.writeValueAsString(((Map) payload.get("itbl")).get("campaignId")));
-                    request.templateId = Integer.parseInt(mapper.writeValueAsString(((Map) payload.get("itbl")).get("templateId")));
-                    request.messageId = (String) ((Map) payload.get("itbl")).get("messageId");
+                    Map<String, Object> iterableMap = (Map) payload.get("itbl");
+                    request.campaignId = convertItblPayloadFieldToInt(iterableMap.get("campaignId"));
+                    request.templateId = convertItblPayloadFieldToInt(iterableMap.get("templateId"));
+                    request.messageId = (String) iterableMap.get("messageId");
+                }
```

Comment:
@vbabenkoru this is what you had in mind right?

Reply:
Yup!

Reply:
I realized we could make this a bit shorter.